File: Code/Source/Estimatorx.Web/app/models/LogEvent.ts
/// <reference path="../_ref.ts" />
module Estimatorx {
  "use strict";

  export interface ILogEvent {
    Id?: string;
    Date?: string;
    Level?: string;
    Logger?: string;
    Message?: string;
    Source?: string;
    Correlation?: string;
    Exception?: ILogError;
    Properties?: any;

    Expanded?: boolean;
  }
}

File: Code/Source/Estimatorx.Core/LogEvent.cs
using System;
using System.Collections.Generic;

namespace Estimatorx.Core
{
    public class LogEvent
    {
        public LogEvent()
        {
            Properties = new Dictionary<string, string>();
        }

        public string Id { get; set; }

        public DateTime Date { get; set; }

        public string Level { get; set; }

        public string Logger { get; set; }

        public string Message { get; set; }

        public string Source { get; set; }

        public string Correlation { get; set; }

        public LogError Exception { get; set; }

        public Dictionary<string, string> Properties { get; set; }
    }
}
